ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -319,6 +319,7 @@ struct Assemble<'a> {
     asm: dynasmrt::x64::Assembler,
     /// Deopt info, with one entry per guard, in the order that the guards appear in the trace.
     deoptinfo: HashMap<usize, (GuardSnapshot, DeoptInfo)>,
+    free_guard_reg: HashMap<usize, Rq>,
```

Comment:
This will definitely need documenting: I'm not really sure what it's doing.

Also, do we need to think about FP registers here?

Reply:
Yeah, I was rushing to make these changes, so used whatever name first came to mind. I'm not sold on this name either.

No, we don't need to worry about FP registers, since we only need one GP register to store the 64-bit jump address.

Reply:
See [d0172f0](https://github.com/ykjit/yk/pull/1591/commits/d0172f0ceafc3e70129eb0a7b7c8060fefb4116e).

To add more context: this is the register that the register allocator frees up for us before we jump to the `fail_label`. We use this register to store the address of the side-trace in order to do 64-bit jumps.